File: SDKs/Aspose_Cloud_SDK_For_PHP/src/Aspose/Cloud/Words/MailMerge.php
<?php
/**
 * Deals with Word document builder aspects.
 */
namespace Aspose\Cloud\Words;

use Aspose\Cloud\Common\AsposeApp;
use Aspose\Cloud\Common\Product;
use Aspose\Cloud\Common\Utils;
use Aspose\Cloud\Exception\AsposeCloudException as Exception;
use Aspose\Cloud\Storage\Folder;

class MailMerge
{

    /*
     * Executes mail merge without regions.
     * 
     * @param string $fileName The source file name.
     * @param string $strXML Data in xml format.
     * 
     * @return string Returns the file path.
     * @throws Exception
     */
    public function executeMailMerge($fileName, $strXML)
    {
        //check whether files are set or not
        if ($fileName == '')
            throw new Exception('File not specified');

        //build URI to execute mail merge without regions
        $strURI = Product::$baseProductUri . '/words/' . $fileName . '/executeMailMerge';

        //sign URI
        $signedURI = Utils::sign($strURI);

        $responseStream = Utils::processCommand($signedURI, 'POST', '', $strXML);
        $json = json_decode($responseStream);

        $v_output = Utils::validateOutput($responseStream);

        if ($v_output === '') {
            //Save docs on server
            $folder = new Folder();
            $outputStream = $folder->GetFile($json->Document->FileName);
            $outputPath = AsposeApp::$outPutLocation . $fileName;
            Utils::saveFile($outputStream, $outputPath);
            return $outputPath;
        } else
            return $v_output;
    }

    /**
     * Executes mail merge with regions.
     *
     * @param string $fileName The name of source file.
     * @param string $strXML Data in xml format.
     *
     * @return string Returns the file path.
     * @throws Exception
     */
    public function executeMailMergewithRegions($fileName, $strXML)
    {
        //check whether files are set or not
        if ($fileName == '')
            throw new Exception('File not specified');

        //build URI to execute mail merge with regions
        $strURI = Product::$baseProductUri . '/words/' . $fileName . '/executeMailMerge?withRegions=true';

        //sign URI
        $signedURI = Utils::sign($strURI);

        $responseStream = Utils::processCommand($signedURI, 'POST', '', $strXML);
        $json = json_decode($responseStream);

        $v_output = Utils::validateOutput($responseStream);

        if ($v_output === '') {
            //Save docs on server
            $folder = new Folder();
            $outputStream = $folder->GetFile($json->Document->FileName);
            $outputPath = AsposeApp::$outPutLocation . $fileName;
            Utils::saveFile($outputStream, $outputPath);
            return $outputPath;
        } else
            return $v_output;
    }

    /**
     * Executes mail merge template.
     *
     * @param string $fileName The name of source file.
     * @param string $strXML Data in xml format.
     * @param string $documentFolder The document folder.
     *
     * @return string Returns the file path.
     * @throws Exception
     */
    public function executeTemplate($fileName, $strXML, $documentFolder = '')
    {
        //check whether files are set or not
        if ($fileName == '')
            throw new Exception('File not specified');

        //build URI to execute mail merge template
        $strURI = Product::$baseProductUri . '/words/' . $fileName . '/executeTemplate' . ($documentFolder == '' ? '' : '?folder=' . $documentFolder);

        //sign URI
        $signedURI = Utils::sign($strURI);

        $responseStream = Utils::processCommand($signedURI, 'POST', '', $strXML);

        $v_output = Utils::validateOutput($responseStream);

        if ($v_output === '') {
            $json = json_decode($responseStream);
            //Save docs on server
            $folder = new Folder();
            $outputStream = $folder->GetFile(($documentFolder == '' ? $json->Document->FileName : $documentFolder . '/' . $json->Document->FileName));
            $outputPath = AsposeApp::$outPutLocation . $fileName;
            Utils::saveFile($outputStream, $outputPath);
            return $outputPath;
        } else
            return $v_output;
    }

}

File: SDKs/Aspose_Cloud_SDK_For_Ruby/lib/Words/mail_merge.rb
module Aspose
  module Cloud
    module Words
      class MailMerge
        def initialize(filename)
          @filename = filename
          raise 'filename not specified.' if filename.empty?
          @base_uri = "#{Aspose::Cloud::Common::Product.product_uri}/words/#{@filename}"
        end
    
=begin
   Executes mail merge with/without regions.
   @param string filename
   @param string str_xml
=end
        def execute_mail_merge(str_xml, with_regions=false, folder_name = '', storage_type = 'Aspose', storage_name = '')
          raise 'str_xml not specified.' if str_xml.empty?

          str_uri = "#{@base_uri}/executeMailMerge"
          str_uri = "#{str_uri}?withRegions=true" if with_regions
          str_uri = Aspose::Cloud::Common::Utils.append_storage(str_uri,folder_name,storage_name,storage_type)
          signed_str_uri = Aspose::Cloud::Common::Utils.sign(str_uri)
          response_stream = RestClient.post(signed_str_uri, str_xml, {:accept=>'application/json'})

          valid_output = Aspose::Cloud::Common::Utils.validate_output(response_stream)
          valid_output.empty? ? Aspose::Cloud::Common::Utils.download_file(JSON.parse(response_stream)['Document']['FileName'],@filename,folder_name,storage_name,storage_type) : valid_output
        end

=begin
  Executes mail merge with template.
  @param string str_xml
=end
        def execute_template(str_xml, folder_name = '', storage_type = 'Aspose', storage_name = '')
          raise 'str_xml not specified.' if str_xml.empty?

          str_uri = "#{@base_uri}/executeTemplate"
          str_uri = Aspose::Cloud::Common::Utils.append_storage(str_uri,folder_name,storage_name,storage_type)
          signed_str_uri = Aspose::Cloud::Common::Utils.sign(str_uri)
          response_stream = RestClient.post(signed_str_uri, str_xml, {:accept=>'application/json'})

          valid_output = Aspose::Cloud::Common::Utils.validate_output(response_stream)
          valid_output.empty? ? Aspose::Cloud::Common::Utils.download_file(JSON.parse(response_stream)['Document']['FileName'],@filename,folder_name,storage_name,storage_type) : valid_output
        end
      end
    end
  end
end